tensordict/nn/distributions/composite.py

```diff
@@ -147,19 +167,71 @@ def rsample(self, shape=None) -> TensorDictBase:
             shape + self.batch_shape,
         )
 
-    def log_prob(self, sample: TensorDictBase) -> TensorDictBase:
-        """Writes a ``<sample>_log_prob entry`` for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
+    def log_prob(self, sample: TensorDictBase) -> torch.Tensor:
+        """Computes and returns the summed log-prob."""
+        if not self.aggregate_probabilities:
+            return self.log_prob_composite(sample, include_sum=True)
+        slp = 0.0
+        for name, dist in self.dists.items():
+            lp = dist.log_prob(sample.get(name))
+            while lp.ndim > sample.ndim:
+                lp = lp.sum(-1)
+            slp = slp + lp
+        return slp
+
+    def log_prob_composite(
+        self, sample: TensorDictBase, include_sum=True
+    ) -> TensorDictBase:
+        """Writes a ``<sample>_log_prob`` entry for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
```

Comment:
This should point to `log_prob` and vice-versa (to let users know what else exists)